primitives/npos-elections/compact/src/lib.rs

```diff
@@ -220,10 +221,59 @@ fn struct_def(
 			pub fn average_edge_count(&self) -> usize {
 				self.edge_count().checked_div(self.len()).unwrap_or(0)
 			}
+
+			/// Remove a certain voter.
+			///
+			/// This will only search until the first instance of `to_remove`, and return true. If
+			/// no instance is found (no-op), then it returns false.
+			///
+			/// In other words, if this return true, exactly one element must have been removed from
+			/// `self.len()`.
+			pub fn remove_voter(&mut self, to_remove: #voter_type) -> bool {
+				#remove_voter_impl
+				return false
+			}
 		}
 	))
 }
 
+fn remove_voter_impl(count: usize) -> TokenStream2 {
+	// 1s
+	let filed_name = field_name_for(1);
+	let single = quote! {
+		if let Some(idx) = self.#filed_name.iter().position(|(x, _)| *x == to_remove) {
+			self.#filed_name.remove(idx);
+			return true
+		}
+	};
+
+	let filed_name = field_name_for(2);
```

Comment:
ditto and bellow too
```suggestion
	let field_name = field_name_for(2);
```